airtable/airtable.py

```diff
@@ -1,582 +1,918 @@
+        """
+
+        all_record_ids = [r["id"] for r in self.get_all(table_name, **options)]
+        deleted_records = self.batch_delete_in_table(table_name, all_record_ids)
+        new_records = self.batch_insert_in_table(table_name, records)
+        return (new_records, deleted_records)
+
+
+class Airtable(AirtableBase):
+
+    def __init__(self, base_key, table_name, api_key=None):
+        """
+        If api_key is not provided, :any:`AirtableAuth` will attempt
+        to use ``os.environ['AIRTABLE_API_KEY']``
+        """
+        super().__init__(base_key, api_key=api_key)
+        self.table_name = table_name
+
+        # the 2 lines below are not really needed but kept just for passing tests
+        # and more importantly for backward comptability
+        # @TODO: breaking change to remove in next release
+        url_safe_table_name = quote(table_name, safe="")
+        self.url_table = posixpath.join(self.API_URL, base_key, url_safe_table_name)
+
+    # same for this function, not really needed too
+    # @TODO: breaking change to add in next release
+
+    def record_url(self, record_id):
+        """ Builds URL with record id """
+        return posixpath.join(self.url_table, record_id)
+
+    def get(self, record_id):
+        """
+        Retrieves a record by its id
+
+        >>> record = airtable.get('recwPQIfs4wKPyc9D')
+
+        Args:
+            record_id(``str``): Airtable record id
+
+        Returns:
+            record (``dict``): Record
+        """
+        return self.get_in_table(self.table_name, record_id)
+
+    def get_iter(self, **options):
+        """
+        Record Retriever Iterator
+
+        Returns iterator with lists in batches according to pageSize.
+        To get all records at once use :any:`get_all`
+
+        >>> for page in airtable.get_iter():
+        ...     for record in page:
+        ...         print(record)
+        [{'fields': ... }, ...]
+
+        Keyword Args:
+            max_records (``int``, optional): The maximum total number of
+                records that will be returned. See :any:`MaxRecordsParam`
+            view (``str``, optional): The name or ID of a view.
+                See :any:`ViewParam`.
+            page_size (``int``, optional ): The number of records returned
+                in each request. Must be less than or equal to 100.
+                Default is 100. See :any:`PageSizeParam`.
+            fields (``str``, ``list``, optional): Name of field or fields to
+                be retrieved. Default is all fields. See :any:`FieldsParam`.
+            sort (``list``, optional): List of fields to sort by.
+                Default order is ascending. See :any:`SortParam`.
+            formula (``str``, optional): Airtable formula.
+                See :any:`FormulaParam`.
+
+        Returns:
+            iterator (``list``): List of Records, grouped by pageSize
+
+        """
+        return self.get_iter_in_table(self.table_name, **options)
```

Comment:
I am not convinced we should modify the method names.

Airtable methods could just super()` call `Airtable` method and pass table name

Reply:
the method name has not been modified in `Airtable`, but just in `AirtableBase`, and it's kind of a refactor to not repeat the same code twice

Reply:
`Airtable` still have the same method name, and I think that for `AirtableBase` adding the `in_table` to each method is better, to make the user aware that it's a method that queries from a table that he should specify

Reply:
I find this pattern confusing, especially because subclass would also have all of the parents class. I would prefer to use method inheritance. It's ok if signatures are different. 
See example.

What do you think?

````
AirtableBase():
    def get(self, table_name, record_id):
        record_url = self.record_url(record_id)
        return self._get(record_url)

Airtable():
    def get(self, record_id):
        return super()._get(self.table_name, record_id)

table = Airtable(x, y, z )
table.get(record_id)

base = AirtableBase(x, y)
base.get(table_name, record_id)
```

Reply:
agree, yes it can be less confusing